src/main/sensors/compass.c

```diff
@@ -388,31 +398,233 @@ uint32_t compassUpdate(timeUs_t currentTimeUs)
     }
 
     flightDynamicsTrims_t *magZero = &compassConfigMutable()->magZero;
-    if (magInit) {              // we apply offset only once mag calibration is done
-        mag.magADC[X] -= magZero->raw[X];
-        mag.magADC[Y] -= magZero->raw[Y];
-        mag.magADC[Z] -= magZero->raw[Z];
-    }
 
     if (tCal != 0) {
         if ((currentTimeUs - tCal) < 30000000) {    // 30s: you have 30s to turn the multi in all directions
             LED0_TOGGLE;
-            for (int axis = 0; axis < 3; axis++) {
-                if (mag.magADC[axis] < magZeroTempMin.raw[axis])
-                    magZeroTempMin.raw[axis] = mag.magADC[axis];
-                if (mag.magADC[axis] > magZeroTempMax.raw[axis])
-                    magZeroTempMax.raw[axis] = mag.magADC[axis];
-            }
+
+            const float recipdTimeSec = 1.0f / (dTimeUs * 1.0e-6f);
+            const float dmag[3] =  {(mag.magADC[X] - magPrevious.magADC[X]) * recipdTimeSec,
```

Comment:
(some vector functions got merged recently. Maybe they would make code  bit more readable and easier to refactor)

Reply:
Thx, I talked to @KarateBrot about this, he kindly revisited my code before the PR. If I would swap to the new vector matrix functionality I would need to rewrite the hole algorithm. If it is ok i would like to leave it as is and postpone the refactoring.

Reply:
Perfectly OK ;-)

The idea was to use it where it is simpler than expanded math + maybe use vector type. But it is better to postpone it until vector code interface stabilizes.